### PR TITLE
chore: adding unit test coverage to `__main__.py`

### PR DESCRIPTION
**Pull request by nick-redfearn**
I originally started this PR with exit codes in mind, but I can see now that this is not needed.

Along the way, I added a bunch of unit test coverage to `__main__.py`. I feel like this is a useful addition for the future so I'm keeping the PR.

### CODE REVIEW

File: tests/unit/test__main.py [hunk 1]
```diff
@@ -35,47 +35,115 @@
...
-    "kube_completions": True,
 }
-CONFIG_RUNNER_ARGS_2 = {
+CONFIG_RUNNER_ARGS_2 = BASE_RUNNER_ARGS | {
+    "config_dir": "gs://pso-kokoro-resources/resources/test/unit/test__main/3validations",
+}
+CONFIG_RUNNER_ARGS_3 = BASE_RUNNER_ARGS | {
+    "config_dir": "gs://pso-kokoro-resources/resources/test/unit/test__main/4partitions",
+}
+CONFIG_RUNNER_ARGS_4 = CONFIG_RUNNER_ARGS_3 | {
```
Comment (nick-redfearn) [inline tests/unit/test__main.py]: Ive made sure not to change the values of `CONFIG_RUNNER_ARGS_1` + `CONFIG_RUNNER_ARGS_2` + `CONFIG_RUNNER_ARGS_3` + `CONFIG_RUNNER_ARGS_4`, Ive used merge syntax to reduce repeated rows.

(I can revert this if its preferred)

**Comment by sundar-mudupalli-work** [comment]
Nick,

Thank you for contributing to the DVT code base. I am not clear specifically the issue you are trying to address. Our preferred approach is to first create an issue that clearly states the problem. Using that description, we should be able to write a test case that demonstrates the problem. We often do that and use that as a means of testing fixes. 

Today, when an exception is raised, a non zero status code is returned. Of course the exception trace is also included. For a tool like DVT which is used primarily by programmers and database admins, returning a non zero status along with the exception trace is sufficient, I think. 

Sundar Mudupalli

**Review by nick-redfearn — COMMENTED** [review]
Thanks for looking @sundar-mudupalli-work 

I've realised since your review that I can achieve what I need using something like this...
```
python -m data_validation configs run -c ./config.yaml || echo "Take some action here"
```

As such, Ive reverted my `__main__.py` changes.

I'm updating the scope of the PR to "chore: adding unit test coverage to `__main__.py`"

**Comment by helensilva14** [comment]
Hi @nick-redfearn! Unit tests failed in our Cloud Build execution, could you please take a look in the error below? Tks! 

```
__________________ ERROR collecting tests/unit/test__main.py ___________________
tests/unit/test__main.py:46: in <module>
    CONFIG_RUNNER_ARGS_1 = BASE_RUNNER_ARGS | {
E   TypeError: unsupported operand type(s) for |: 'dict' and 'dict'
```

**Comment by nick-redfearn** [comment]
Thanks @helensilva14 - I think its because Python 3.8 doesnt support the | operand type, oops

Ive pushed this commit, I think it should be fine now -> https://github.com/GoogleCloudPlatform/professional-services-data-validator/pull/1329/commits/15b09907da4942de06aa5e3dc7475f1d5f199bc6

File: tests/unit/test__main.py (hunk 1 above)
Comment (nick-redfearn) [reply tests/unit/test__main.py]: Reverted here -> https://github.com/GoogleCloudPlatform/professional-services-data-validator/pull/1329/commits/15b09907da4942de06aa5e3dc7475f1d5f199bc6

**Comment by helensilva14** [comment]
/gcbrun

All good @nick-redfearn, thanks for the timely fix! Starting build again and let's see if all will pass!

File: tests/unit/test__main.py [hunk 2]
```diff
@@ -207,3 +327,194 @@ def test_config_runner_4(mock_args, mock_build, mock_run, caplog):
...
+    "data_validation.__main__.build_config_managers_from_args",
+    return_value=[
+        config_manager.ConfigManager(
+            VALIDATE_COLUMN_CONFIG, MockIbisClient(), MockIbisClient(), verbose=False
+        )
+    ],
+)
+@mock.patch("data_validation.__main__.run_validation")
+def test_successful_column_validation_with_mocked_run_validation(
+    mock_args, mock_build, mock_run
```
Comment (nj1973) [inline tests/unit/test__main.py:346]: Either these arguments in are in the wrong order of the `mock.patch` calls are in the wrong order. I don't suppose it matters in practice as long as they do what they do. But I guess the names should match up.

File: tests/unit/test__main.py [hunk 3]
```diff
@@ -207,3 +327,194 @@ def test_config_runner_4(mock_args, mock_build, mock_run, caplog):
...
+            BROKEN_VALIDATE_COLUMN_CONFIG_MISSING_COMMAND,
+            MockIbisClient(),
+            MockIbisClient(),
+            verbose=False,
+        )
+    ],
+)
+@mock.patch("data_validation.__main__.run_validation")
+def test_throws_for_malformed_input_config_missing_command(
+    mock_args, mock_build, mock_run
```
Comment (nj1973) [inline tests/unit/test__main.py:368]: Arg order.

File: tests/unit/test__main.py [hunk 4]
```diff
@@ -207,3 +327,194 @@ def test_config_runner_4(mock_args, mock_build, mock_run, caplog):
...
+            BROKEN_VALIDATE_COLUMN_CONFIG_INCORRECT_COMMAND,
+            MockIbisClient(),
+            MockIbisClient(),
+            verbose=False,
+        )
+    ],
+)
+@mock.patch("data_validation.__main__.run_validation")
+def test_throws_for_malformed_input_config_incorrect_command(
+    mock_args, mock_build, mock_run
```
Comment (nj1973) [inline tests/unit/test__main.py:392]: Arg order.
I won't add any more of these because it's probably annoying. I'm sure you can find the rest yourself :-)

**Review by nj1973 — CHANGES_REQUESTED** [review]
Broadly speaking, looks good. Just a minor comment regarding mock order.

File: tests/unit/test__main.py (hunk 2 above)
Comment (nick-redfearn) [reply tests/unit/test__main.py:346]: @nj1973 
I dont see whats wrong, can you suggest what order the `mock_args, mock_build, mock_run` should be in. Im happy to make the change throughout the script

Comment (nj1973) [reply tests/unit/test__main.py:346]: I can't find it in the pytest docs but args are in reverse order to the `@mock.patch` calls.

I wasn't aware of this myself but noticed some other tests in the same file had them reversed, for example:
```
@mock.patch("data_validation.__main__.run_validations")
@mock.patch(
    "data_validation.__main__.build_config_managers_from_yaml",
    return_value=["config dict from one file"],
)
@mock.patch(
    "argparse.ArgumentParser.parse_args",
    return_value=argparse.Namespace(**CONFIG_RUNNER_ARGS_4),
)
def test_config_runner_4(mock_args, mock_build, mock_run, caplog):
```

So I added a breakpoint in your new test and had a look at the contents of the parameters. `mock_args` contained `@mock.patch("data_validation.__main__.run_validation")`.

It doesn't really matter in practice but I thought we should be consistent.

Comment (nick-redfearn) [reply tests/unit/test__main.py:346]: @nj1973 Thanks for explaining, I wasn't aware of this detail.

Ill make a fix now 👍

Comment (nick-redfearn) [reply tests/unit/test__main.py:346]: @nj1973 Fixed here -> https://github.com/GoogleCloudPlatform/professional-services-data-validator/pull/1329/commits/796783564a28997b77cd0f4d689c73892139314a

File: tests/unit/test__main.py (hunk 3 above)
Comment (nick-redfearn) [reply tests/unit/test__main.py:368]: Fixed here -> https://github.com/GoogleCloudPlatform/professional-services-data-validator/pull/1329/commits/796783564a28997b77cd0f4d689c73892139314a

File: tests/unit/test__main.py (hunk 4 above)
Comment (nick-redfearn) [reply tests/unit/test__main.py:392]: Fixed here -> https://github.com/GoogleCloudPlatform/professional-services-data-validator/pull/1329/commits/796783564a28997b77cd0f4d689c73892139314a

**Review by nj1973 — APPROVED** [review]
LGTM, thanks again for contributing.

**Pull request merged**